src/Tools/ExternalAccess/AspNetCore/Internal/EmbeddedLanguages/AspNetCoreEmbeddedLanguageDocumentHighlighter.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.DocumentHighlighting;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.Internal.EmbeddedLanguages
+{
+    [ExportEmbeddedLanguageDocumentHighlighter(
+        nameof(AspNetCoreEmbeddedLanguageDocumentHighlighter),
+        new[] { LanguageNames.CSharp },
+        supportsUnannotatedAPIs: false,
+        // Add more syntax names here in the future if there are additional cases ASP.Net would like to light up on.
+        identifiers: new[] { "Route" }), Shared]
+    internal class AspNetCoreEmbeddedLanguageDocumentHighlighter : IEmbeddedLanguageDocumentHighlighter
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public AspNetCoreEmbeddedLanguageDocumentHighlighter()
+        {
+        }
+
+        public ImmutableArray<DocumentHighlights> GetDocumentHighlights(
+            Document document,
+            SemanticModel semanticModel,
+            SyntaxToken token,
+            int position,
+            HighlightingOptions options,
+            CancellationToken cancellationToken)
+        {
+            var highlighters = AspNetCoreDocumentHighlighterExtensionProvider.GetExtensions(document.Project);
+            foreach (var highlighter in highlighters)
```

Comment:
do we ever expect there to be more than one aspnet implementation for each of the embedded language features?

Reply:
probably.... not.  but this just follows the same pattern for *everything* asp.net.... so just nicer to keep consistent :D